### PR TITLE
Add EcoWitt WH31B support by modifying WH31E

### PR DESCRIPTION
**Pull request by bitfliq**
Added samples here: https://github.com/merbanan/rtl_433_tests/pull/394 , https://github.com/bitfliq/rtl_433_tests/tree/EcoWitt-WH31B .  

This code worked for extracting data from WH31B - but I was unable to verify that it did not break the other devices.

Thanks for all of your hard work on this fantastic project.

### CODE REVIEW

**Comment by bitfliq** [comment]
Closing due to instability. Will reopen after fixes in place.

**Comment by zuckschwerdt** [comment]
Notes:
- leave the crc-error distinction out, maybe add `(%d) ... msg_type`
- leave long/short id out, use:
- `                    "model",            "",             DATA_COND, msg_type == 0x30, DATA_STRING, "AmbientWeather-WH31E",`
   `                    "model",            "",             DATA_COND, msg_type == 0x37, DATA_STRING, "AmbientWeather-WH31B",`

**Comment by bitfliq** [comment]
Thanks @zuckschwerdt for friendly feedback.  Instability seems resolved with your suggestions.  Not confident I matched what you were looking for with crc-error distinction recommendation.

**Comment by zuckschwerdt** [comment]
Looks mostly good, thanks! I meant to say: just change the error strings, e.g. "WH31E bad CRC" to "WH31E/WH31B bad CRC", there is no real need to have a switch on the msg_type there (just `%d`, like you did) -- shorter code is more legible :)

Perhaps we should keep CRC/SUM separate, so e.g. `fprintf(stderr, "%s: WH31E/WH31B (%s) bad CRC\n", __func__, msg_type);`

**Comment by bitfliq** [comment]
@zuckschwerdt Could you please take another look?

File: src/devices/ambientweather_wh31e.c [hunk 1]
```diff
@@ -189,27 +192,26 @@ static int ambientweather_whx_decode(r_device *decoder, bitbuffer_t *bitbuffer)
...
-            fprintf(stderr, "%s: WH31E/WH40 detected, buffer is %d bits length\n", __func__, bitbuffer->bits_per_row[row]);
+            fprintf(stderr, "%s: WH31E/WH31B/WH40 detected, buffer is %d bits length\n", __func__, bitbuffer->bits_per_row[row]);
 
         // remove preamble, keep whole payload
         bitbuffer_extract_bytes(bitbuffer, row, start_pos + 24, b, 18 * 8);
         msg_type = b[0];
 
-        if (msg_type == 0x30) {
-            // WH31E
+        if (b[0] == wh31e_type_code || b[0] == wh31b_type_code) {
```
Comment (Mindavi) [inline src/devices/ambientweather_wh31e.c:201]: Use msg_type here, rather than b[0].

File: src/devices/ambientweather_wh31e.c [hunk 2]
```diff
@@ -189,27 +192,26 @@ static int ambientweather_whx_decode(r_device *decoder, bitbuffer_t *bitbuffer)
...
             }
             uint8_t c_sum = add_bytes(b, 6) - b[6];
             if (c_sum) {
                 if (decoder->verbose)
-                    fprintf(stderr, "%s: WH31E bad SUM\n", __func__);
+                    fprintf(stderr, "%s: WH31E/WH31B (%d) bad SUM\n", __func__, msg_type);
                 continue; // DECODE_FAIL_MIC
             }
-
+
```
Comment (Mindavi) [inline src/devices/ambientweather_wh31e.c:214]: Whitespace error?

File: src/devices/ambientweather_wh31e.c (hunk 1 above)
Comment (bitfliq) [reply src/devices/ambientweather_wh31e.c]: Thanks, changed.

File: src/devices/ambientweather_wh31e.c (hunk 2 above)
Comment (bitfliq) [reply src/devices/ambientweather_wh31e.c]: Thanks, I believe this is fixed.

**Pull request merged**